crates/key-derivation/src/operator.rs

```diff
@@ -0,0 +1,453 @@
+    }
+}
+
+// Manual Drop implementation to zeroize keys on drop.
+impl Drop for OperatorKeys {
+    fn drop(&mut self) {
+        #[cfg(feature = "zeroize")]
+        self.zeroize();
+    }
+}
+
+#[cfg(feature = "zeroize")]
+impl Zeroize for OperatorKeys {
+    #[inline]
+    fn zeroize(&mut self) {
+        let Self {
+            master,
+            base,
+            message,
+            wallet,
+        } = self;
+
+        // # Security note
+        //
+        // Going over all possible "zeroizable" fields.
+        // What we cannot zeroize is only:
+        //
+        // - Network: enum
+        //
+        // These are fine to leave as they are since they are public parameters,
+        // and not secret values.
+        //
+        // NOTE: `Xpriv.private_key` (`SecretKey`) `non_secure_erase` writes `1`s to the memory.
+
+        // Zeroize master components
+        master.depth.zeroize();
+        {
+            let fingerprint: &mut [u8; 4] = master.parent_fingerprint.as_mut();
+            fingerprint.zeroize();
+        }
+        master.private_key.non_secure_erase();
+        {
+            let chaincode: &mut [u8; 32] = master.chain_code.as_mut();
+            chaincode.zeroize();
+        }
+        let raw_ptr = &mut master.child_number as *mut ChildNumber;
+        // SAFETY: `master.child_number` is a valid enum variant
+        //          and will not be accessed after zeroization.
+        //          Also there are only two possible variants that will
+        //          always have an `index` which is a `u32`.
+        //          Note that `ChildNumber` does not have the `#[non_exhaustive]`
+        //          attribute.
+        unsafe {
+            *raw_ptr = if master.child_number.is_normal() {
+                ChildNumber::Normal { index: 0 }
+            } else {
+                ChildNumber::Hardened { index: 0 }
+            };
+        }
+
+        // Zeroize base components
+        base.depth.zeroize();
+        {
+            let fingerprint: &mut [u8; 4] = base.parent_fingerprint.as_mut();
+            fingerprint.zeroize();
+        }
+        base.private_key.non_secure_erase();
+        {
+            let chaincode: &mut [u8; 32] = base.chain_code.as_mut();
+            chaincode.zeroize();
+        }
+        let raw_ptr = &mut base.child_number as *mut ChildNumber;
+        // SAFETY: `base.child_number` is a valid enum variant
+        //          and will not be accessed after zeroization.
+        //          Also there are only two possible variants that will
+        //          always have an `index` which is a `u32`.
+        //          Note that `ChildNumber` does not have the `#[non_exhaustive]`
+        //          attribute.
+        unsafe {
+            *raw_ptr = if base.child_number.is_normal() {
+                ChildNumber::Normal { index: 0 }
+            } else {
+                ChildNumber::Hardened { index: 0 }
+            };
+        }
```

Comment:
In theory the compiler could optimize this in the absence of an optimization barrier. But the consequences of this seem pretty trivial.

Reply:
Yes, but even with the `#[inline]`?

Reply:
It still could, to my knowledge. But is `ChildNumber` strictly secret data? I agree that a best-effort approach should still be taken so the user gets from `Zeroize` what the name implies; but I wouldn't worry much about this particular case.

Reply:
No, it is a public parameter and shown in the open source code in `strata`.

Reply:
Got it. Then it's definitely fine (and on further thought, could even be ignored with a suitable comment on the `zeroize` implementation if you really wanted).